code/modules/events/scrubber_clog.dm

```diff
@@ -0,0 +1,144 @@
+				)
+			return pick(major_mobs)
+
+		if("Critical") //Higher impact mobs, but with a lower max spawn.
+			var/list/critical_mobs = list(
+				/mob/living/simple_animal/hostile/carp,
+				/mob/living/simple_animal/hostile/bee/toxin,
+				/mob/living/basic/cockroach/glockroach,
+				)
+			return pick(critical_mobs)
+
+		if("Strange") //Useful or silly mobs. Still hazardous. Very low weight.
+			var/list/strange_mobs = list(
+				/mob/living/simple_animal/hostile/retaliate/goose, //Janitors HATE geese.
+				/mob/living/simple_animal/hostile/bear,
+				/mob/living/simple_animal/pet/gondola,
+				/mob/living/simple_animal/hostile/mushroom,
+				/mob/living/simple_animal/hostile/lightgeist
+				)
+			return pick(strange_mobs)
+
+/datum/round_event/scrubber_clog/start() //Sets the scrubber up for unclogging/mob production
+	SEND_SIGNAL(scrubber, COMSIG_VENT_CLOG)
+	SEND_SIGNAL(scrubber, COMSIG_PRODUCE_MOB, spawned_mob, living_mobs) //The first one's free!
+
+/datum/round_event/scrubber_clog/tick() //Checks if spawn_interval is met, then sends signal to scrubber to produce a mob
+	if(activeFor % spawn_delay == 0 && scrubber.is_clogged() == TRUE)
+		life_check()
+		if(living_mobs.len < maximum_spawns && clogged)
+			SEND_SIGNAL(scrubber, COMSIG_PRODUCE_MOB, spawned_mob, living_mobs)
+
+/datum/round_event/scrubber_clog/end() //No end announcement. If you want to take the easy way out and just leave the vent welded, you must open it at your own peril
+	SEND_SIGNAL(scrubber, COMSIG_VENT_UNCLOG)
+
+/datum/round_event/scrubber_clog/proc/life_check()
+	for(var/mob/living/mob_check in living_mobs)
+		if(mob_check.health <= 0 || !mob_check.health)
+			living_mobs -= mob_check
+
+/datum/round_event_control/scrubber_clog/major
+	name = "Major Scrubber Clog"
+	typepath = /datum/round_event/scrubber_clog/major
+	weight = 12
+	max_occurrences = 3
+	earliest_start = 10 MINUTES
+
+/datum/round_event/scrubber_clog/major
+	severity = "Major"
+
+/datum/round_event/scrubber_clog/major/setup()
+	. = ..()
+	maximum_spawns = rand(2,4)
+	spawn_delay = rand(15,20)
+
+/datum/round_event_control/scrubber_clog/critical
+	name = "Critical Scrubber Clog"
+	typepath = /datum/round_event/scrubber_clog/critical
+	weight = 8
+	min_players = 15
+	max_occurrences = 1
+	earliest_start = 25 MINUTES
+
+/datum/round_event/scrubber_clog/critical
+	severity = "Critical"
+
+/datum/round_event/scrubber_clog/critical/setup()
+	. = ..()
+	maximum_spawns = 3
+	spawn_delay = rand(15, 25)
+
+/datum/round_event/scrubber_clog/critical/announce()
+	priority_announce("Potentially hazardous lifesigns detected in the [get_area(scrubber)] ventilation network.", "Security Alert")
+
+/datum/round_event_control/scrubber_clog/strange
+	name = "Strange Scrubber Clog"
+	typepath = /datum/round_event/scrubber_clog/strange
+	weight = 5
+	max_occurrences = 1
+
+/datum/round_event/scrubber_clog/strange
+	severity = "Strange"
+
+/datum/round_event/scrubber_clog/strange/setup()
+	. = ..()
+	endWhen = rand(6000,8000)
```

Comment:
time defines

Reply:
![image](https://user-images.githubusercontent.com/28870487/168923478-2bf23828-f77d-48ef-9c16-ae2ae0761666.png)

the "activefor" timer that gets used for starting/ending/etc only counts up once a second, not in deciseconds. Using timedefines for some of the stuff in here will make it last ten times longer than intended. I'm gonna go back and undo some of the defines I added, but I just wanted to give a reason why first.